content/doc/book/using/jenkins-best-practices.adoc

```diff
@@ -0,0 +1,97 @@
+This practice also implies that all code including third-party jars, build scripts, release notes, etc. must be checked into Source Code Control.
+
+✅Integrate tightly with your issue tracking system, like JIRA or bugzilla, to reduce the need for maintaining a Change Log.
+
+The integration helps to track changes as they are made, including build status, what build has been performed for this requirement or defects, and the link to the actual build results and artifacts.
+
+✅Integrate tightly with a repository browsing tool like FishEye if you are using Subversion as source code management tool.
+
+Repository browsing provides a quick update on what happens on a Subversion repository.
+It also provides a graphical diff on what changes have been made from the previous build.
+
+✅Always configure your job to generate trend reports and automated testing when running a Java build.
+
+Trends helps project managers and developers quickly visualize current project progress status.
+Moreover, unit testing is often not enough to provide confidence that the delivered software complies to the desired quality.
+The more you test the software, the better the delivered software complies to the desired quality.
+
+✅Set up Jenkins on the partition that has the most free disk-space.
+
+Jenkins needs some disk space to perform builds and keep archives. All the settings, build logs, artifact archives are stored under the JENKINS_HOME directory.
+Simply archive this directory to make a backup.
+Similarly, restoring the data is just replacing the contents of the JENKINS_HOME directory from a backup.
+
+✅Archive unused jobs before removing them.
+
+All unused jobs should be archived, so they can be resurrected if the need arises.
+See link:/doc/book/system-administration/administering-jenkins[Administering Jenkins] for ways to do this.
+
+✅Setup a different job/project for each maintenance or development branch you create.
+
+One of advantages of using CI tools is to detect problems early in the development lifecycle.
+Setting up a different job/project for each branch you create will help to maximize the benefit of detecting problems early as part of supporting parallel development efforts and reducing risk.
+
+✅Prevent resource collisions in jobs that are running in parallel.
+
+Multiple jobs running at the same time often cause collisions if they set up some kind of service, or need exclusive access.
+If your builds involve use of databases or other networked services, you need to ensure that they don't interfere with each other.
+Allocate a different port for parallel project builds to avoid build collisions.
+If that's not possible (e.g. in the case of a persistent resource that needs to be locked) you can prevent builds that use it from running at the same time using e.g. link:https://github.com/jenkinsci/throttle-concurrent-builds-plugin[Throttle Concurrent Builds Plugin.]
```

Comment:
```suggestion
✅Prevent resource collisions in jobs that are running in parallel.

Multiple jobs running at the same time often cause collisions if they set up some kind of service, or need exclusive access.
If your builds involve the use of databases or other networked services, you need to ensure that they don't interfere with each other.
Allocate a different port for parallel project builds to avoid build collisions.
If that's not possible (e.g. in the case of a persistent resource that needs to be locked) you can prevent builds that use it from running at the same time using e.g. link:https://github.com/jenkinsci/throttle-concurrent-builds-plugin[Throttle Concurrent Builds Plugin.]
```